src/framework/visualization/generateVisualization.ts

```diff
@@ -0,0 +1,34 @@
+import path from 'path';
+import { getRootStorageDirectory, writeFileToPath } from '../../fileSystem';
+import { generateVizHTML } from './generateVizHTML';
+import { retrieveIntegrationData } from './retrieveIntegrationData';
+
+/**
+ * Generates visualization of Vertices and Edges using https://visjs.github.io/vis-network/docs/network/
+ */
+export async function generateVisualization(): Promise<string> {
+  const integrationPath = getRootStorageDirectory();
+
+  const { entities, relationships } = await retrieveIntegrationData(
+    integrationPath,
+  );
+
+  const nodeDataSets = entities.map((entity) => ({
+    id: entity._key,
+    label: entity.name.toString(),
+  }));
+  const edgeDataSets = relationships.map((relationship) => ({
+    from: relationship._fromEntityKey.toString(),
+    to: relationship._toEntityKey.toString(),
+    label: relationship.displayName.toString(),
+  }));
```

Comment:
Aren't these fields already typed as string? I don't think the extra safety of `.toString()` is necessary. Its also missing on the `_key` above for the `nodeDataSets`.

Reply:
Talked with @charlieduong94 there was an issue with the typing and I just needed to be more specific with my types to avoid needing to cast. Good catch!